core/src/test/scala/unit/kafka/zk/KafkaZkClientTest.scala

```diff
@@ -1057,6 +1057,51 @@ class KafkaZkClientTest extends ZooKeeperTestHarness {
     assertEquals(Set(topicPartition10, topicPartition11), zkClient.getAllPartitions)
   }
 
+  /**
+   * Create a large number of topic names, such that their total (aggregate) length is more than 1 MB,
+   * and verify that our ZK client can still read them all despite being configured for 1 MB max response
+   * size.
+   *
+   * NOTE THAT THIS LOOPS INDEFINITELY if paginateTopics = false, so a negative test is currently not
```

Comment:
If paginateTopic=false, GetChildrenRequest would be called. Does this mean there is a bug when calling LiZookeeper GetChildrenRequest?

Reply:
Correct. More precisely, the bug is in our ZooKeeperClient class's handling of the error-response. Arguably there's also a bug in Zookeeper itself in that it apparently uses a SessionExpiredException (retryable) instead of an explicitly non-retryable exception (jute.maxbuffer isn't going to fix itself no matter how many times you retry.)